Pilot/pilotCommands.py

```diff
@@ -484,27 +474,21 @@ def execute(self):
 
     # If NumberOfProcessors or MaxRAM are defined in the resource configuration, these
     # values are preferred
+
+    # pilotProcessors is basically the number of processors this pilot is "managing"
+    self.pp.pilotProcessors = numberOfProcessorsOnWN
+
+    # payloadProcessors is the number of processors used by the single payloads.
+    # We store payloadProcessors in the global parameters so that other commands can more easily use it.
+    # (MultiLaunchAgent is right now is the only consumer)
+    self.pp.payloadProcessors = 1
     if "WholeNode" in self.pp.tags:
-      self.pp.payloadProcessors = numberOfProcessorsOnWN
-    if self.pp.maxNumberOfProcessors > 0:
-      self.pp.payloadProcessors = min(numberOfProcessorsOnWN, self.pp.maxNumberOfProcessors)
-
-    if not self.pp.payloadProcessors:
-      self.log.warn("Could not retrieve number of processors, assuming 1")
-      self.pp.payloadProcessors = 1
-
-    # Make sure the multiprocessor tags are present if not already there
-    if self.pp.payloadProcessors > 1:
-      if 'MultiProcessor' not in self.pp.tags:
-        self.pp.tags.append('MultiProcessor')
-      if 'MultiProcessor' not in self.pp.reqtags:
-        self.pp.reqtags.append('MultiProcessor')
-      if ('%dProcessors' % self.pp.payloadProcessors) not in self.pp.tags:
```

Comment:
After my PR the other week I realised that GridPP has 8Processors (or whatever) as the required tag and MultiProcessor as a normal tag. It would be ok to remove the tags part here and apply them from the dirac-pilot command line in the submitted job script or in the VM user_data.